core/formatter.cpp

```diff
@@ -1724,16 +1733,233 @@ class FixIndentation {
+    }
+
+    void sortGroup(ImportElems &imports) {
+        // We don't want to change behaviour in such case:
+        // local foo = "b.jsonnet";
+        // local foo = "a.jsonnet";
+        // So we don't change the order when there are shadowed variables.
+        if (!duplicatedVariables(imports)) {
+            std::sort(imports.begin(), imports.end());
+        }
+    }
+
+    ImportElems extractImportElems(Local *local, Fodder after) {
+        ImportElems result;
+        for (auto &bind: local->binds) {
+            Import *import = dynamic_cast<Import*>(bind.body);
+            assert(import != nullptr);
+            result.emplace_back(sortingKey(import), after, &bind);
+            after = {FodderElement(FodderElement::Kind::LINE_END, 0, 0, {})};
+        }
+        return result;
+    }
 
+    AST *buildGroupAST(ImportElems &imports, AST *body, const Fodder &groupOpenFodder) {
+        for (int i = imports.size() - 1; i >= 0; --i) {
+            auto &import = imports[i];
+            Fodder fodder;
+            if (i == 0) {
+                fodder = groupOpenFodder;
+            } else {
+                fodder = imports[i - 1].afterFodder;
+            }
+            Local::Binds binds({*import.bind});
+            auto *local = alloc.make<Local>(
+                LocationRange(),
+                fodder,
+                binds,
+                body
+            );
+            body = local;
+        }
 
-// TODO(dcunnin): Add pass to alphabeticize top level imports.
+        return body;
+    }
 
+    bool duplicatedVariables(const ImportElems &elems) {
+        std::vector<UString> idents;
+        for (const auto &elem: elems) {
+            idents.push_back(elem.bind->var->name);
+        }
+        return std::unique(idents.begin(), idents.end()) != idents.end();
+    }
+
+    /// Check if the import group ends after this local
+    bool groupBoundaryAfter(Local *local) {
+        Local *next = goodLocalOrNull(local->body);
+        if (!next) {
+            return true;
+        }
+
+        bool newlineReached = false;
+        for (const auto &fodderElem: open_fodder(next)) {
+            if (newlineReached || fodderElem.blanks > 0) {
+                return true;
+            }
+            if (fodderElem.kind != FodderElement::Kind::INTERSTITIAL) {
+                newlineReached = true;
+            }
+        }
+        return false;
+    }
+
+    void ensureCleanNewline(Fodder &fodder) {
+        if (fodder.size() == 0 || fodder.back().kind == FodderElement::Kind::INTERSTITIAL) {
+            fodder.push_back(FodderElement(FodderElement::Kind::LINE_END, 0, 0, {}));
+        }
+    }
+
+    AST *toplevelImport(Local *local, ImportElems &imports, const Fodder &groupOpenFodder) {
+        assert(isGoodLocal(local));
+
+        Fodder commentAfter, beforeNextFodder;
+        std::tie(commentAfter, beforeNextFodder) = splitOpenFodder(local->body);
+
+        ensureCleanNewline(commentAfter);
```

Comment:
Maybe commentAfter is not the best name. Any suggestions? It contains all comments until the end of the line. In case a comment goes beyond the end of the line it's also a part of commentAfter. However if there are some blank lines afterwards, they go to beforeNextFodder.

Example: 
```
local foo = 'a.jsonnet'; /* blah blah*/ /* blah blah
blah blah

*/

// something else

{}
```
commentAfter =
```
/* blah blah*/ /* blah blah
blah blah

*/
```
beforeNextFodder ==
```

// something else

```